src/averageAges.js

```diff
@@ -55,7 +82,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(child => {
+      const isMother = people
+        .some((woman) => child.mother === woman.name);
```

Comment:
you do this twice, move this to the separate fucntion
![image](https://user-images.githubusercontent.com/91411692/233588717-651051a5-71c4-4b73-9d8f-f7265428c070.png)